launcher/src/main/scripts/bin/launcher.py

```diff
@@ -71,7 +71,9 @@ def try_lock(f):
 
 def open_read_write(f, mode):
     """Open file in read/write mode (without truncating it)"""
-    return os.fdopen(os.open(f, O_RDWR | O_CREAT, mode), 'r+')
+    fd = os.open(f, O_RDWR | O_CREAT, mode)
+    os.dup2(os.dup(fd), fd)  # backward compatible approach to make it inheritable (PEP 446)
```

Comment:
I would prefer following, but let's wait for @electrum opinion too
```
if hasattr(os, 'set_inheritable'):
    # File descriptor returned from os.open is not inheritable since Python 3.4
    os.set_inheritable(fd, True)
```

i would also rename `open_read_write` to `open_pidfile`, because the mechanics of the function are adjusted for this use-case

Reply:
@antonymayi would that work for you?

Reply:
sure, I agree it is much more readable, perhaps I was bit too minimalistic...

Reply:
PR updated according to this suggestion